lib/ansible/executor/task_executor.py

```diff
@@ -401,18 +401,17 @@ def _execute(self, variables=None):
             # the options specified on the command line
             self._play_context = self._play_context.set_task_and_variable_override(task=self._task, variables=variables, templar=templar)
 
-            # fields set from the play/task may be based on variables, so we have to
-            # do the same kind of post validation step on it here before we use it.
-            self._play_context.post_validate(templar=templar)
-
-            # now that the play context is finalized, if the remote_addr is not set
-            # default to using the host's address field as the remote address
+            # if the remote_addr is not set, default to using the host's address field
             if not self._play_context.remote_addr:
                 self._play_context.remote_addr = self._host.address
 
             # We also add "magic" variables back into the variables dict to make sure
             # a certain subset of variables exist.
             self._play_context.update_vars(variables)
```

Comment:
some of the magic vars rely on post_validate already being run to have the correct value